runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java

```diff
@@ -296,4 +310,62 @@ private void drawTooltip(Graphics2D graphics, XpGlobe mouseOverSkill, int goalXp
 
 		xpTooltip.render(graphics);
 	}
+
+	private void drawLevels(Graphics2D graphics, int startXp, int goalXp, int x, int y)
+	{
+		final Collection<Double> levelPercentages = levelPercentagesCache.computeIfAbsent(
+			Pair.of(startXp, goalXp),
+			unused -> Experience.levelPercentagesBetweenExperience(startXp, goalXp)
+		);
+
+
+		final Stroke backupStroke = graphics.getStroke();
+		graphics.setStroke(new BasicStroke(2));
+
+		for (final double levelMarker : levelPercentages)
+		{
+			final double angle = Math.toRadians(levelMarker * 360);
+			final Vector4<Double> coordinates = calculateCoordinatesForAngle(angle, x, y);
+
+			drawLine(
+				graphics,
+				coordinates.getX(),
+				coordinates.getY(),
+				coordinates.getZ(),
+				coordinates.getW()
+			);
+		}
+
+		graphics.setStroke(backupStroke);
+	}
+
+
+	private Vector4<Double> calculateCoordinatesForAngle(double angle, double x, double y)
+	{
+		final int size = config.xpOrbSize();
+		final double radius = config.xpOrbSize() / 2f;
+
+		final double middleX = x + (size / 2f);
+		final double middleY = y + (size / 2f);
+
+		// Start coordinates
+		final double x1 = (radius - (config.progressArcStrokeWidth() / 2f)) * Math.sin(angle);
+		final double y1 = (radius - (config.progressArcStrokeWidth() / 2f)) * -Math.cos(angle);
+
+		// End coordinates
+		final double x2 = (radius + (config.progressArcStrokeWidth() / 2f)) * Math.sin(angle);
+		final double y2 = (radius + (config.progressArcStrokeWidth() / 2f)) * -Math.cos(angle);
+
+		return new Vector4<>(middleX + x1, middleY + y1, middleX + x2, middleY + y2);
+	}
+
+	private void drawLine(Graphics2D graphics, double x1, double y1, double x2, double y2)
+	{
+		graphics.drawLine(
+			(int) Math.round(x1),
```

Comment:
Can this be done without converting to and from doubles?

Reply:
The x and y variables here are always rational numbers , because they are the coordinates of a point on the circle. We use Math.round(double) to get it more pixel perfect with the circle.